packages/core/src/clients/create-public-client.ts

```diff
@@ -0,0 +1,22 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate'
+import type { Prettify } from '../types/utils.js'
+import { type Client, type ClientConfig, createClient } from './create-client'
+import { type PublicActions, publicActions } from './decorators/public'
+
+export type PublicClientConfig = ClientConfig & {
+  cosmWasmClient: CosmWasmClient
+}
+
+export type PublicClient = Prettify<Client<PublicActions>>
+
+export function createPublicClient(
```

Comment:
This decorator pattern is fascinating and I haven't seen it in action before. I see it mirrors viem 👀